tap/passive_tapper.go

```diff
@@ -129,32 +131,38 @@ func printPeriodicStats(cleaner *Cleaner) {
 	}
 }
 
-func startPassiveTapper(outputItems chan *api.OutputChannelItem) {
-	streamsMap := NewTcpStreamMap()
-	go streamsMap.closeTimedoutTcpStreamChannels()
-
-	diagnose.InitializeErrorsMap(*debug, *verbose, *quiet)
-	diagnose.InitializeTapperInternalStats()
-
+func initializePacketSource() (*source.PacketSourceManager, error) {
 	var bpffilter string
 	if len(flag.Args()) > 0 {
 		bpffilter = strings.Join(flag.Args(), " ")
 	}
 
-	packetSource, err := source.NewTcpPacketSource(*fname, *iface, source.TcpPacketSourceBehaviour{
+	behaviour := source.TcpPacketSourceBehaviour{
 		SnapLength:  *snaplen,
 		Promisc:     *promisc,
 		Tstype:      *tstype,
 		DecoderName: *decoder,
 		Lazy:        *lazy,
 		BpfFilter:   bpffilter,
-	})
+	}
+
+	return source.NewPacketSourceManager(*procfs, *pids, *fname, *iface, behaviour)
+}
+
+func startPassiveTapper(outputItems chan *api.OutputChannelItem) {
+	streamsMap := NewTcpStreamMap()
+	go streamsMap.closeTimedoutTcpStreamChannels()
+
+	diagnose.InitializeErrorsMap(*debug, *verbose, *quiet)
+	diagnose.InitializeTapperInternalStats()
+
+	sources, err := initializePacketSource()
```

Comment:
the name is little confusing. because the variable is "sources" (multiple) and the name of the function is "PacketSource" (single).

Also, in the `initializePacketSource` it looks like we are actually starting source manager (NewPacketSourceManager) so maybe there is some name changes needed here..

Reply:
Correct, fixing in